spec/unit/ssl/state_machine_spec.rb

```diff
@@ -820,7 +820,8 @@ def write_csr_attributes(data)
             csr.custom_attributes
           ).to contain_exactly(
                  {'oid' => '1.3.6.1.4.1.34380.1.2.1', 'value' => 'CSR specific info'},
-                 {'oid' => '1.3.6.1.4.1.34380.1.2.2', 'value' => 'more CSR specific info'}
+                 {'oid' => '1.3.6.1.4.1.34380.1.2.2', 'value' => 'more CSR specific info'},
+                 {'oid' => 'pp_auth_auto_renew', 'value' => 'true'}
```

Comment:
Could you try referencing the oid  as `1.3.6.1.4.1.34380.1.3.2` instead of the "short name"?

Reply:
ah nvm, it looks like the JRuby implementation is buggy when it comes to CSR attributes which is why we disable all of the CSR generation tests in https://github.com/puppetlabs/puppet/blob/fde57136980d63bd2a64532ad74f750cc3b73bc7/spec/unit/ssl/certificate_request_spec.rb#L72

I'd do the same for the failing test. We don't ever generate a CSR within a JRuby/puppetserver context, so it can be ignored.

Reply:
Ahhh okay, got it!